### PR TITLE
Adding update duties script

### PR DESCRIPTION
**Pull request by Shurc1val**
Added folder `update_duties`, containing two files:
- `update_duties.py` contains functions to be run (daily?) to update duty table in db against current API data.
- `database_functions.py`contains functions interacting with the database that are used in the file `update_duties.py`

Tested against live db and api with a couple of routine cases; not documented yet.

### CODE REVIEW

File: update_duties/update_duties.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        print(f'Checking duty for plant with id {plant_id}...')
+        if len(botanist_name) == 2:
+            botanist_id = dbf.get_botanist_by_name(botanist_name[0], botanist_name[1], database)[0]
+            if not check_for_existing_duty(plant_id, botanist_id):
+                dbf.add_duty(database, plant_id, botanist_id)
+
+
+if __name__ == "__main__":
+    load_dotenv()
+
```
Comment (Shurc1val) [inline update_duties/update_duties.py:52]: double space here is annoying...

File: update_duties/database_functions.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+"""Module to contain """
```
Comment (caitlinturnidge) [inline update_duties/database_functions.py]: Finish the docstring

File: update_duties/database_functions.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+from datetime import datetime
+from os import environ
+
+from dotenv import load_dotenv
+import pandas as pd
+import sqlalchemy as db
+
+
+class MSSQL_Database():
```
Comment (caitlinturnidge) [inline update_duties/database_functions.py:12]: Nice use of a class!

File: update_duties/update_duties.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import database_functions as dbf
+
+
+BASE_URL = "https://data-eng-plants-api.herokuapp.com/plants/"
+
+
+def get_api_botanist_name_by_plant_id(plant_id: int) -> list[dict]:
+    """Gets plant data from API using ID."""
+    api_botanist_info = requests.get(BASE_URL + str(plant_id),
```
Comment (caitlinturnidge) [inline update_duties/update_duties.py]: small thing - if you import 'get' from requests then you can just write get() when you call the API

File: update_duties/database_functions.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+def get_table_records(table_name: str, database: MSSQL_Database) -> pd.DataFrame:
+    """Retrieves all records as pandas df from specified table in db."""
+    try:
+        table = db.Table(table_name, database.metadata, autoload_with=database.engine)
+        query = db.select(table)
+        response = database.connection.execute(query)
+        results = response.fetchall()
+        return pd.DataFrame(results)
+
+    except Exception as e:
```
Comment (caitlinturnidge) [inline update_duties/database_functions.py:116]: Do we definitely want to raise these errors? could it cause the script to stop running at any point?

File: update_duties/update_duties.py (hunk 4 above)
Comment (Shurc1val) [reply update_duties/update_duties.py]: good point

File: update_duties/database_functions.py (hunk 5 above)
Comment (Shurc1val) [reply update_duties/database_functions.py:116]: It depends on when it's used.... this function isn't actually used in the file `update_duties.py`, it's a legacy from a small refactor I made, but with the view of eventually expanding this file to be a library of database functions, I thought it was useful enough to leave in.

**Review by caitlinturnidge — APPROVED** [review]
Looks great

**Pull request merged**